solidity/contracts/hooks/libs/AbstractMessageIdAuthHook.sol

```diff
@@ -82,20 +86,27 @@ abstract contract AbstractMessageIdAuthHook is
             metadata.msgValue(0) < 2 ** 255,
             "AbstractMessageIdAuthHook: msgValue must be less than 2 ** 255"
         );
-        bytes memory payload = abi.encodeCall(
-            AbstractMessageIdAuthorizedIsm.verifyMessageId,
-            id
-        );
-        _sendMessageId(metadata, payload);
+
+        _sendMessageId(metadata, message);
+
+        uint256 _overpayment = msg.value - _quoteDispatch(metadata, message);
+        if (_overpayment > 0) {
+            address _refundAddress = metadata.refundAddress(msg.sender);
+            require(
+                _refundAddress != address(0),
+                "AbstractPostDispatchHook: no refund address"
+            );
+            payable(_refundAddress).sendValue(_overpayment);
```

Comment:
[In `LayerZeroV2Hook.sol`](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/9093369186993468571b5d411e712ddd9a30c98c/solidity/contracts/hooks/layer-zero/LayerZeroV2Hook.sol#L81), when calling `lZEndpoint.send{value: msg.value}(msgParams, refundAddress);`, all of `msg.value` is sent instead of the amount specified by `_quoteDispatch()`, which can cause a revert in overpayment scenarios

Reply:
Fix with `lZEndpoint.send{value: quote}(msgParams, refundAddress)`